Possible malformed POM on mkl javacpp-preset

Investigating a download failure on the pom for the mkl library in javacpp-presets. This does not seem to be a network issue, and yet the jar is available here:

https://oss.sonatype.org/content/repositories/snapshots/org/bytedeco/javacpp-presets/mkl/2017.2-1.3.3-SNAPSHOT/
https://oss.sonatype.org/content/repositories/snapshots/org/bytedeco/javacpp-presets/mkl/2017.2-1.3.4-SNAPSHOT/

I managed to reduce it to:

```
mvn -U org.apache.maven.plugins:maven-dependency-plugin:2.2:get     -DrepoUrl=https://oss.sonatype.org/content/repositories/snapshots/     -Dartifact=org.bytedeco.javacpp-presets:mkl:jar:2017.2-1.3.3-SNAPSHOT
```

Run in an empty directory, this command fails with:
```
[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Building Maven Stub Project (No POM) 1
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-dependency-plugin:2.2:get (default-cli) @ standalone-pom ---
Downloading: https://repo.maven.apache.org/maven2/org/bytedeco/javacpp-presets/mkl/jar/mkl-jar.2017.2-1.3.3-SNAPSHOT
Downloading: https://oss.sonatype.org/content/repositories/snapshots/org/bytedeco/javacpp-presets/mkl/jar/mkl-jar.2017.2-1.3.3-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 2.026 s
[INFO] Finished at: 2017-09-08T21:58:20+02:00
[INFO] Final Memory: 19M/1472M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-dependency-plugin:2.2:get (default-cli) on project standalone-pom: Couldn't download artifact: Could not find artifact org.bytedeco.javacpp-presets:mkl:2017.2-1.3.3-SNAPSHOT:jar in central (https://repo.maven.apache.org/maven2)
[ERROR] 
[ERROR] Try downloading the file manually from the project website.
[ERROR] 
[ERROR] Then, install it using the command: 
[ERROR]     mvn install:install-file -DgroupId=org.bytedeco.javacpp-presets -DartifactId=mkl -Dversion=jar -Dpackaging=2017.2-1.3.3-SNAPSHOT -Dfile=/path/to/file
[ERROR] 
[ERROR] Alternatively, if you host your own repository you can deploy the file there: 
[ERROR]     mvn deploy:deploy-file -DgroupId=org.bytedeco.javacpp-presets -DartifactId=mkl -Dversion=jar -Dpackaging=2017.2-1.3.3-SNAPSHOT -Dfile=/path/to/file -Durl=[url] -DrepositoryId=[id]
[ERROR] 
[ERROR] 
[ERROR]   org.bytedeco.javacpp-presets:mkl:2017.2-1.3.3-SNAPSHOT:jar
[ERROR] 
[ERROR] from the specified remote repositories:
[ERROR]   central (https://repo.maven.apache.org/maven2, releases=true, snapshots=false),
[ERROR]   temp (https://oss.sonatype.org/content/repositories/snapshots/, releases=true, snapshots=true)
[ERROR] 
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

This is relevant because:

```
huitseeker@sleipnir➜huitseeker/DL4J/nd4j(master✗)» ag -B2 -A3 '2-1\.3\.3'


nd4j-backends/nd4j-backend-impls/nd4j-native-platform/pom.xml
12-    <properties>
13-        <openblas.version>0.2.19</openblas.version>
14:        <mkl.version>2017.2-1.3.3-SNAPSHOT</mkl.version>
15-        <nd4j.backend>nd4j-native</nd4j.backend>
16-    </properties>
17-


nd4j-backends/nd4j-backend-impls/nd4j-native/pom.xml
12-    <properties>
13-        <openblas.version>0.2.19</openblas.version>
14:        <mkl.version>2017.2-1.3.3-SNAPSHOT</mkl.version>
15-        <env.LIBND4J_HOME>${basedir}/../../../../libnd4j/</env.LIBND4J_HOME>
16-    </properties>
17-

```

/cc @saudet

--- a/nd4j-backends/nd4j-backend-impls/nd4j-native-platform/pom.xml
+++ b/nd4j-backends/nd4j-backend-impls/nd4j-native-platform/pom.xml
@@ -11,7 +11,7 @@
 
     <properties>
         <openblas.version>0.2.19</openblas.version>
-        <mkl.version>2017.2-1.3.3-SNAPSHOT</mkl.version>
+        <mkl.version>2017.3</mkl.version>
         <nd4j.backend>nd4j-native</nd4j.backend>
     </properties>
 
@@ -20,6 +20,11 @@
             <groupId>org.bytedeco.javacpp-presets</groupId>
             <artifactId>openblas-platform</artifactId>
             <version>${openblas.version}-${javacpp-presets.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.bytedeco.javacpp-presets</groupId>
+            <artifactId>mkl-platform</artifactId>
+            <version>${mkl.version}-${javacpp-presets.version}</version>
         </dependency>
         <dependency>
             <groupId>${project.groupId}</groupId>

--- a/nd4j-backends/nd4j-backend-impls/nd4j-native/pom.xml
+++ b/nd4j-backends/nd4j-backend-impls/nd4j-native/pom.xml
@@ -11,7 +11,7 @@
 
     <properties>
         <openblas.version>0.2.19</openblas.version>
-        <mkl.version>2017.2-1.3.3-SNAPSHOT</mkl.version>
+        <mkl.version>2017.3</mkl.version>
         <env.LIBND4J_HOME>${basedir}/../../../../libnd4j/</env.LIBND4J_HOME>
     </properties>
 
@@ -42,13 +42,13 @@
         <dependency>
             <groupId>org.bytedeco.javacpp-presets</groupId>
             <artifactId>mkl</artifactId>
-            <version>${mkl.version}</version>
-            <classifier>linux-x86_64</classifier>
+            <version>${mkl.version}-${javacpp-presets.version}</version>
         </dependency>
         <dependency>
             <groupId>org.bytedeco.javacpp-presets</groupId>
             <artifactId>mkl</artifactId>
-            <version>${mkl.version}</version>
+            <version>${mkl.version}-${javacpp-presets.version}</version>
+            <classifier>${dependency.platform}</classifier>
         </dependency>
 
         <dependency>

